src/Symfony/Component/Security/Http/Firewall/ContextListener.php

```diff
@@ -225,6 +226,12 @@ protected function refreshUser(TokenInterface $token)
                 $this->logger->debug('Token was deauthenticated after trying to refresh it.');
             }
 
+            if (null !== $this->dispatcher) {
+                $this->dispatcher->dispatch(new UserHasChangedEvent($token, $user), UserHasChangedEvent::class);
+
+                return $token;
```

Comment:
I mean _only_ in case it has been reauthenticated :)

```php
if ($token->isAuthenticated()) {
   return $token;
}
```

Reply:
understood !